rebuild/src/addon.h

```diff
@@ -0,0 +1,177 @@
+#ifndef BLST_TS_ADDON_H__
+#define BLST_TS_ADDON_H__
+
+#include <openssl/rand.h>
+
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <string_view>
+
+#include "blst.hpp"
+#include "napi.h"
+
+// TODO: these should come out post PR review
+using std::cout;
+using std::endl;
+
+#define BLST_TS_RANDOM_BYTES_LENGTH 8U
+
+#define BLST_TS_FUNCTION_PREAMBLE(info, env, module)                           \
+    Napi::Env env = info.Env();                                                \
+    Napi::EscapableHandleScope scope(env);                                     \
+    BlstTsAddon *module = env.GetInstanceData<BlstTsAddon>();
+
+#define BLST_TS_IS_INFINITY                                                \
+    Napi::Env env = info.Env();                                                \
+    Napi::EscapableHandleScope scope(env);                                     \
+    return scope.Escape(Napi::Boolean::New(env, _point->IsInfinite()));
+
+#define BLST_TS_SERIALIZE_POINT(macro_name)                                \
+    Napi::Env env = info.Env();                                                \
+    Napi::EscapableHandleScope scope(env);                                     \
+    bool compressed{true};                                                     \
+    if (!info[0].IsUndefined()) {                                              \
```

Comment:
Can `info` be empty?

Reply:
no. its passed to all JS function invocations by the runtime.